test/when_all_range_test.cpp

```diff
@@ -0,0 +1,129 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#include <gtest/gtest.h>
+
+#include <unifex/allocate.hpp>
+#include <unifex/any_sender_of.hpp>
+#include <unifex/finally.hpp>
+#include <unifex/get_stop_token.hpp>
+#include <unifex/just.hpp>
+#include <unifex/just_from.hpp>
+#include <unifex/let_value_with_stop_source.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/sync_wait.hpp>
+#include <unifex/then.hpp>
+
+#include <unifex/never.hpp>
+#include <unifex/when_all_range.hpp>
+
+using namespace unifex;
+
+class WhenAllRangeTests : public ::testing::Test {};
+
+namespace {
+constexpr auto times_three = [](int x) noexcept {
+  return then(just(x), [](int val) noexcept { return val * 3; });
+};
+}  // namespace
+
+TEST_F(WhenAllRangeTests, givenReceiver_whenAllValue_thenReceivedValue) {
+  // ----- WHEN -----
```

Comment:
Let's delete these comments; they seem to be idiomatic in the source we're upstreaming from, but they stand out as unusual in this repo.
```suggestion
```